modules/ding_place2book/ding_place2book.fields.inc

```diff
@@ -1 +1,298 @@
+  $event = $event_id = $event_maker_id = $price_id = NULL;
+  if (!empty($items)) {
+    list($event_id, $event_maker_id, $price_id) = array_values(reset($items));
+    try {
+      $p2b = P2b\P2b::getInstance();
+      $event = $p2b->getEvent($event_maker_id, $event_id);
+      $price = $p2b->getPrice($event_maker_id, $event_id, $price_id);
+    }
+    catch (Exception $ex) {
+      watchdog_exception('ding_place2book', $ex, t('Error appeared on getting data from p2b.'));
+    }
+  }
+  switch ($instance['widget']['type']) {
+    case 'ding_p2b_widget_default':
+      $element['place2book'] = [
+        '#type' => 'fieldset',
+        '#title' => t('Place2Book'),
+        '#description' => t('Information about event on Place2book service.'),
+        '#collapsible' => TRUE,
+        '#collapsed' => TRUE,
+        '#tree' => TRUE,
+      ];
+
+      $element['place2book']['synchronize'] = [
+        '#type' => 'checkbox',
+        '#title' => t('Synchronize event with Place2book service'),
+        '#description' => t('If checked, then event on Place2book will be updated with data from local event.'),
+        '#default_value' => variable_get('p2b_default_syncronize', FALSE),
+      ];
+
+      if (!empty($event)) {
+        $begin_at = date('d/m/Y H:i:s', strtotime($event->begin_at));
+        $end_at = date('d/m/Y H:i:s', strtotime($event->end_at));
+        $element['place2book'] += [
+          'name' => ['#markup' => "<p><strong>Name:</strong> <span>{$event->name}</span></p>"],
+          'description_short' => ['#markup' => "<p><strong>Short description:</strong> <span>{$event->description_short}</span></p>"],
+          'description_long' => ['#markup' => "<p><strong>Long description:</strong> <span>{$event->description_long}</span></p>"],
+          'begin_at' => ['#markup' => "<p><strong>Begin at:</strong> <span>{$begin_at}</span></p>"],
+          'end_at' => ['#markup' => "<p><strong>End at:</strong> <span>{$end_at}</span></p>"],
+          'price' => ['#markup' => "<p><strong>End at:</strong> <span>{$price->value}</span></p>"],
+        ];
+
+        if (!empty($price)) {
+          $element['place2book']['price']['#markup'] = "<p><strong>Event price:</strong> <span>{$price->value}</span></p>";
+        }
+
+        if (!empty($event->address)) {
+          $address = implode(', ', (array) $event->address);
+          $element['place2book']['address']['#markup'] = "<p><strong>Sale open at:</strong> <span>{$address}</span></p>";
+        }
+
+        if (!empty($event->sale_open_at)) {
+          $sale_open_at = date('d/m/Y H:i:s', strtotime($event->sale_open_at));
+          $element['place2book']['sale_open_at']['#markup'] = "<p><strong>Sale open at:</strong> <span>{$sale_open_at}</span></p>";
+        }
+
+        if (!empty($event->sale_close_at)) {
+          $sale_close_at = date('d/m/Y H:i:s', strtotime($event->sale_close_at));
```

Comment:
Could be a problem to change the date format.
Suggest to use internal Drupal date formats.